src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8238,6 +8318,45 @@ internal static bool WinIsSameFileSystemItem(string pathOne, string pathTwo)
             return false;
         }
 
+        internal static bool GetInodeData(string path, out System.ValueTuple<UInt64, UInt64> inodeData)
+        {
```

Comment:
I see that you are following a pattern like `IsSameFileSystemItem` or `IsHardLink`. Ideally, code here should be refactored to consolidate Unix/Windows implementations in one place and eliminate wrapper functions like `IsSameFileSystemItem` and `IsHardLink`. Instead they should be
```CSharp
internal static bool IsSameFileSystemItem(string pathOne, string pathTwo)
{
#if UNIX
   // Unix implementation goes here
#else
   // Windows implementation goes here
#endif
}
```
I'm not asking you to do this change for this PR, just want to call out that it should be this way moving forward. I opened #4086 to track the refactoring task.